src/AutoHashEqualsCached.jl

```diff
@@ -37,25 +54,27 @@ function _show_default_auto_hash_equals_cached(io::IO, @nospecialize(x))
 end
 
 # Find the first struct declaration buried in the Expr.
-get_struct_decl(typ) = nothing
-function get_struct_decl(typ::Expr)
+get_struct_decl(__source__, typ) = nothing
+function get_struct_decl(__source__, typ::Expr)
     if typ.head === :struct
         return typ
     elseif typ.head === :macrocall
-        return get_struct_decl(typ.args[3])
+        return get_struct_decl(__source__, typ.args[3])
     elseif typ.head === :block
         # get the first struct decl in the block
         for x in typ.args
-            if x.head === :macrocall || x.head === :struct || x.head === :block
-                result = get_struct_decl(x)
+            if x isa LineNumberNode
+                __source__ = x
+            elseif x isa Expr && (x.head === :macrocall || x.head === :struct || x.head === :block)
```

Comment:
Not a big deal, but is it worth abstracting these away into helper functions to make this easier to understand?

I'm not fully sure what the different forms mean `:macrocall`, `:struct`, etc., so might be nice to comment them or put them in a function like `represents_struct`. Alternatively, `try_get_struct_decl` which returns `nothing` if invalid, a not-nothing `result` if it is.